modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java

```diff
@@ -0,0 +1,629 @@
+/*
+ * Copyright (C) 2021 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.impl.volatility.smile;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.joda.beans.Bean;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.value.ValueDerivatives;
+import com.opengamma.strata.collect.array.DoubleArray;
+
+/**
+ * Adjustments to the SABR parameters to accommodate the pricing of in-arrears caplets.
+ * <p>
+ * The parameter "q" in the formula is a parameter such that 1-q is closely related to a mean reversion.
+ * It is defaulted to q=1 in the default instance of the formula.
+ * <p>
+ * Reference: Willems, Sander. SABR smiles for RFR caplets. August 2020. 
+ * Electronic copy available at: https://ssrn.com/abstract=3567655
+ */
+@BeanDefinition(factoryName = "of")
+public final class SabrInarrearsVolatilityFunction
+    implements ImmutableBean, Serializable {
+
+  /** The default parameter for q, with 1-q related to mean-reversion. The default
+   * Correspond to no mean reversion in the in-arrears period. */
+  private static final double DEFAULT_Q = 1.0;
+
+  /** The mean reversion related parameter. */
+  @PropertyDefinition
+  private final double q;
+
+  /**
+   * Default implementation with q = 1;
+   */
+  public static final SabrInarrearsVolatilityFunction DEFAULT = new SabrInarrearsVolatilityFunction(DEFAULT_Q);
+
+  /**
+   * The effective SABR parameters from the raw SABR parameters and the times. Formula in the case tau0 > 0.
+   * <p>
+   * Theorem 4.2 in reference.
+   * 
+   * @param parameters  the raw SABR parameters
+   * @param tau0  the accumulation period start time
+   * @param tau1  the accumulation period end time
+   * @return  the effective SABR parameters
+   */
+  public SabrFormulaData effectiveSabrBeforeStart(SabrFormulaData parameters, double tau0, double tau1) {
+    double alpha = parameters.getAlpha();
+    double beta = parameters.getBeta();
+    double rho = parameters.getRho();
+    double nu = parameters.getNu();
+    double tau = 2 * q * tau0 + tau1;
+    double tau2 = tau * tau;
```

Comment:
Maybe use underscore to show it's squared i.e tau1_2 and tau0_2. Do you think its clearer what is subscript and what is a power?

Reply:
Good idea, as we have tau1 and tau2 with different meaning of the number.

Reply:
Actually, stylecheck does not like the '_' in the names. I reverted to the old names.